spring-vault-core/src/main/java/org/springframework/vault/support/AbstractResult.java

```diff
@@ -85,8 +85,8 @@ public V get() {
 		if (isSuccessful()) {
 			return get0();
 		}
-
-		throw new VaultException(exception.getMessage());
+		// Mark: This seemed wrong, and lost information
```

Comment:
The proper implementation would be throwing a completion exception similar to a `Future`. Let's keep this one out of the pull request and address the issue in a separate ticket.